internal/provider/resource_tfe_data_retention_policy.go

```diff
@@ -0,0 +1,368 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-framework-validators/numbervalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/objectvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/numberplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"strings"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &resourceTFEDataRetentionPolicy{}
+var _ resource.ResourceWithConfigure = &resourceTFEDataRetentionPolicy{}
+var _ resource.ResourceWithImportState = &resourceTFEDataRetentionPolicy{}
+
+func NewDataRetentionPolicyResource() resource.Resource {
+	return &resourceTFEDataRetentionPolicy{}
+}
+
+// resourceTFEDataRetentionPolicy implements the tfe_data_retention_policy resource type
+type resourceTFEDataRetentionPolicy struct {
+	config ConfiguredClient
+}
+
+func (r *resourceTFEDataRetentionPolicy) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_data_retention_policy"
+}
+
+func (r *resourceTFEDataRetentionPolicy) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages the data retention policies for a specific workspace or an the entire organization.",
+		Version:     1,
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "ID of the Data Retention Policy.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"organization": schema.StringAttribute{
+				Description: "Name of the organization. If omitted, organization must be defined in the provider config.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
```

Comment:
Would it be possible to add a validation that either organization or workspace is specified?
Edit: Or maybe both? I see that both is possible now...I feel like it would be acceptable to say just one or the other should be specified, to reduce ambiguity. But I dont feel strongly

Reply:
I left in it because I saw that seemed to be the pattern for some other resources, but I think you're right that it makes a lot more sense this way 👍